### DISCUSSION TITLE
SX1262 set/get for OutputPower missing get API?

### DISCUSSION THREAD

**Discussion by MrSniffer** [jgromes/RadioLib#616] [Ideas]
Perhaps I missed it, but in searching the headers for the SX1262, I do see a Set and Get for CurrentLimit, but only a Set for OutputPower. 

Is there a way to return the OutputPower setting that is in effect?

**Comment by jgromes** [comment]
Currently, there are not "getter" methods for most of the configuration parameters like frequency or output power, there was just no need to add them. The get method for current limit was needed to implement power configuration.

However, recently there was a pull request that added some getter methods for CC1101 and RF69 (https://github.com/jgromes/RadioLib/pull/614), so if there is use case for other modules, that could be added as well.

**Reply by MrSniffer** [reply]
My field deployed emergency communications network, is periodically polled for connectivity and signal strength between all neighbors. It is intended to adjust settings, such as power and current, whether to relay or not, etc. in order to maintain connectivity despite failing devices while minimizing power usage during brownouts. Devices can fail for multiple causes; this system is targeted for communications during a wildfire emergency, wherein relay points will just burn down. Additional relays will need to be brought online and power settings adjusted for the new configuration.

Also, as part of general health monitoring, each device is polled and reports myriad statistics and settings, many of which can be changed remotely by admins. Initial conditions for power and current are kept in EEPROM on the devices and not known to rebooted admin stations, hence the need for GETs for all SETs.

My project interest is solely focused on SX1262-based radios.

**Reply by jgromes** [reply]
> My field deployed emergency communications network

I am getting slightly nervous as to the amount of emergency/disaster communication networks that seem to be using RadioLib...

Agree that as a rule of thumb, if there's a set, there should be a get. That's a bit of an antipattern/laziness in my design. I'll create a feature request issue so that I can better keep track of it.

EDIT: Tracking in #622

**Reply by MrSniffer** [reply]
Thanks for the intention to add this. No pressure, being part of critical resource management & monitoring systems....it's a credit to your great contribution!!!!
